pkg/check/chunkrepair/chunkrepair.go

```diff
@@ -0,0 +1,281 @@
+	"github.com/prometheus/common/expfmt"
+)
+
+// Options represents chunk repair check options
+type Options struct {
+	NumberOfChunksToRepair int
+	Seed                   int64
+}
+
+// Check ...
+func Check(c bee.Cluster, o Options, pusher *push.Pusher, pushMetrics bool) (err error) {
+	ctx := context.Background()
+	rnds := random.PseudoGenerators(o.Seed, o.NumberOfChunksToRepair)
+	fmt.Printf("Seed: %d\n", o.Seed)
+
+	pusher.Collector(repairedCounter)
+	pusher.Collector(repairedTimeGauge)
+	pusher.Collector(repairedTimeHistogram)
+
+	pusher.Format(expfmt.FmtText)
+
+	for i := 0; i < o.NumberOfChunksToRepair; i++ {
+		// Pick node A, B, C and a chunk which is closest to B
+		nodeA, nodeB, nodeC, chunk, err := getNodes(ctx, c, rnds[i])
+		if err != nil {
+			return err
+		}
+		addressA, err := nodeA.Overlay(ctx)
+		if err != nil {
+			return err
+		}
+
+		// upload the chunk in nodeA
+		err = nodeA.UploadChunk(ctx, chunk)
+		if err != nil {
+			return err
+		}
+
+		for {
+			// check if the node is there in the local store of node B
+			// this does a get chunk instead of Has chunk, so the following
+			// call just checks if the chunk is accessible from nodeB
+			present, err := nodeB.HasChunk(ctx, chunk.Address())
+			if err != nil {
+				fmt.Println("sleeping...")
+
+				// give time for the chunk to reach its destination
+				time.Sleep(100 * time.Millisecond)
+				continue
+			}
+
+			if present {
+				break
+			}
+		}
+
+		// download the chunk from nodeC
+		data1, err := nodeC.DownloadChunk(ctx, chunk.Address(), "")
+		if err != nil {
+			return err
+		}
+		if !bytes.Equal(data1, chunk.Data()) {
+			return errors.New("chunk downloaded in NodeC does not have proper data")
+		}
+
+		// delete the chunk from all nodes. If the chunk from nodeA is not deleted,
+		// it is hard to simulate the chunk failure in small clusters. We would need a
+		// fairly large cluster then.
+		err = deleteChunkFromAllNodes(ctx, c, chunk)
+		if err != nil {
+			return err
+		}
+
+		// trigger downloading of the chunk from nodeC again (this time it should trigger chunk repair)
+		_, err = nodeC.DownloadChunk(ctx, chunk.Address(), addressA.String()[0:2])
+		if err == nil { // for status Accepted (209), a error is returned
+			return err
+		}
+
+		// by the time the NodeC creates a trojan chunk and asks NodeA to repair, upload the
+		// original chunk in nodeA and pin it
+		err = uploadAndPinChunkToNode(ctx, &nodeA, chunk)
+		if err != nil {
+			return err
+		}
+		time.Sleep(5 * time.Second) // give sometime so that the repair happens
+
+		// download again to see if the chunk is repaired
+		data3, err := nodeC.DownloadChunk(ctx, chunk.Address(), "")
+		if err != nil { // this time it should succeed
+			return err
+		}
+		if !bytes.Equal(data3, chunk.Data()) {
+			return errors.New("chunk downloaded in NodeC does not have proper data")
+		}
+		fmt.Println("repaired chunk ", chunk.Address().String())
+
+		if pushMetrics {
+			if err := pusher.Push(); err != nil {
```

Comment:
None of the metrics are changed during this test. Basically they are not used. This push call will always push the same initial values of all metrics.